ONSdigital/rm-collection-exercise-service
# Improve exception handling and logging for SDS

# What and why?
This PR improves the exception handling and logging of the SDS consumer. It introduces a new resource not found exception for when the SDS message doesn't return a valid collection exercise. It also improves the logging so the period and survey id of the message is logged and an explicit message is displayed when the collection exercise cannot be found. 
# How to test?
Publish a message to your topic:
```
gcloud pubsub topics publish projects/ras-rm-dev/topics/supplementary-data-service-<YOUR_NAMESPACE> --message='{
      "survey_id": "009",
      "period_id": "2025",
      "form_types": [
         "0001"
        ],
      "title": "MBS",
      "sds_published_at": "2023-08-24T11:55:00Z",
      "total_reporting_units": 10,
      "schema_version": "v1.0.0",
      "sds_dataset_version": 3,
      "filename": "815e0310-91a9-48d4-82a6-e827966421d2.json",
      "dataset_id": "815e0310-91a9-48d4-82a6-e827966421d2"
      }'
```
Check the log output
Deploy this PR to your environment. 
Publish a message to your topic and check the logs for the updated log information. Should work for null values as well.
# Jira
https://jira.ons.gov.uk/secure/RapidBoard.jspa?rapidView=1493&view=detail&selectedIssue=RAS-1083
